core/src/test/scala/org/apache/spark/MapOutputTrackerSuite.scala

```diff
@@ -171,6 +200,36 @@ class MapOutputTrackerSuite extends SparkFunSuite {
     mapWorkerRpcEnv.shutdown()
   }
 
+  test("remote get all map output statuses with metadata") {
```

Comment:
I am thinking about extending the `remote fetch` test (the one before this) with an extra registered map output where metadata is given and then this test could be deleted.
 
That way you will test the case when one of map status is given with and one is without a metadata.

WDYT?

Reply:
The previous test has `masterTracker.unregisterMapOutput` and some test verification for that, thus want to avoid adding too much for that test. Also this test is specifically testing non-null metadata object, kind of following "separation of concerns" to make it as a separate test.

Reply:
fine for me